src/php/globals.rs

```diff
@@ -24,4 +32,17 @@ impl ExecutorGlobals {
 
         unsafe { ZendHashTable::from_ptr(self.class_table, false) }.ok()
     }
+
+    pub fn take_exception() -> Option<Box<ZendObject>> {
+        let globals = Self::get_mut();
+
+        let mut exception_ptr = std::ptr::null_mut();
+        std::mem::swap(&mut exception_ptr, &mut globals.exception);
+
+        if !exception_ptr.is_null() {
+            Some(unsafe { Box::from_raw(exception_ptr) })
```

Comment:
As I understand the php source code, we're responsible for freeing it if we take it.